codegen/method.go

```diff
@@ -622,73 +654,113 @@ func getQueryMethodForType(typeSpec compile.TypeSpec) string {
 	case *compile.StringSpec:
 		queryMethod = "GetQueryValue"
 	default:
-		panic(fmt.Sprintf("Unknown type (%T) %v", typeSpec, typeSpec))
+		panic(fmt.Sprintf(
+			"Unknown type (%T) %v for query string parameter",
+			typeSpec, typeSpec,
+		))
 	}
 
 	return queryMethod
 }
 
 func (ms *MethodSpec) setQueryParamStatements(
-	funcSpec *compile.FunctionSpec,
+	funcSpec *compile.FunctionSpec, packageHelper *PackageHelper,
 ) error {
 	// If a thrift field has a http.ref annotation then we
 	// should not read this field from query parameters.
 	statements := LineBuilder{}
-	structType := compile.FieldGroup(funcSpec.ArgsSpec)
 
-	for _, field := range structType {
+	var finalError error
+
+	visitor := func(
+		goPrefix string, thriftPrefix string, field *compile.FieldSpec,
+	) bool {
 		realType := compile.RootTypeSpec(field.Type)
-		fieldName := field.Name
-		identifierName := camelCase(fieldName) + "Query"
+		longFieldName := goPrefix + "." + pascalCase(field.Name)
+
+		// If the type is a struct then we cannot really do anything
+		if _, ok := realType.(*compile.StructSpec); ok {
+			// if a field is a struct then we must do a nil check
+
+			typeName, err := GoType(packageHelper, realType)
+			if err != nil {
+				finalError = err
+				return true
+			}
+
+			statements.appendf("if requestBody%s == nil {", longFieldName)
+			statements.appendf("\trequestBody%s = &%s{}",
+				longFieldName, typeName,
+			)
+			statements.append("}")
+
+			return false
+		}
+
+		var longQueryName string
+		if thriftPrefix == "" {
+			longQueryName = field.Name
+		} else {
+			longQueryName = thriftPrefix + "." + field.Name
+			if longQueryName[0] == '.' {
```

Comment:
Is this nested `if` necessary? can it be treated as a case for the outer `if` so that we don't need nested if?